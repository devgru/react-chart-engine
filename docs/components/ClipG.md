# ClipG

Renders `g` element, used to clip chart contents. Prevents chart contents from 'overflow'.

## Children

Children will be rendered in `g` element, clipped by rectangular `clip-path` with `width` and `height` dimensions.

## Props

All props are **required**.

- string `id`, unique identifier;
- number `width`, chart contents width;
- number `height`, chart contents height.

## Side effects

`ClipG` renders `clip-path` element, identified by `id` being equal to `id` property. This `clip-path` can be used to clip other page elements.
