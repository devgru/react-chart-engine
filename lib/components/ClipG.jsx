import React from 'react';
import PropTypes from 'prop-types';

function ClipG({ id, width, height, children }) {
  return (
    <g>
      <defs>
        <clipPath id={`${id}.clip`}>
          <rect width={width} height={height} />
        </clipPath>
      </defs>
      <g clipPath={`url(#${id}.clip)`}>
        {children}
      </g>
    </g>
  );
}

ClipG.propTypes = {
  children: PropTypes.node,

  id: PropTypes.string.isRequired,
  width: PropTypes.number.isRequired,
  height: PropTypes.number.isRequired,
};

ClipG.defaultProps = {
  children: null,
};

export default ClipG;
