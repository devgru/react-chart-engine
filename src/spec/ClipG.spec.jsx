import React from 'react';
import renderer from 'react-test-renderer';

import ClipG from '../lib/components/ClipG';

describe('ClipG', () => {
  it('should work with props', () => {
    const component = renderer.create(
      <ClipG id='clip' width={100} height={100}>
        <circle/>
      </ClipG>);
    expect(component.toJSON()).toMatchSnapshot();
  });
});
