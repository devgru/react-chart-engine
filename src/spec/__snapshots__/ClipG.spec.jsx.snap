// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ClipG should work with props 1`] = `
<g>
  <defs>
    <clipPath
      id="clip"
    >
      <rect
        height={100}
        width={100}
      />
    </clipPath>
  </defs>
  <g
    clipPath="url(#clip)"
  >
    <circle />
  </g>
</g>
`;
